src/cwe_checker_lib/src/intermediate_representation/macros.rs

```diff
@@ -0,0 +1,549 @@
+///     assert_eq!(
+///     expr!("RAX:8"),
+///     Expression::Var(Variable {name: "RAX".into(), size: ByteSize::new(8),is_temp: false})
+///     );
+///
+///     assert_eq!(
+///     expr!("RAX:8 + 0x42:8"),
+///     Expression::BinOp { op: BinOpType::IntAdd,
+///         lhs: Box::new(Expression::Var(Variable { name: "RAX".into(), size: ByteSize::new(8), is_temp: false })),
+///         rhs: Box::new(Expression::Const(Bitvector::from_u8(0x42)))}
+///     );
+/// ```
+#[macro_export]
+macro_rules! expr {
+    (  $x:expr  ) => {
+        parsing::parse_expr($x)
+    };
+}
+
+/// Creates a `Vec<Term<Def>>` specified by the string slices. Utilizes `variable!`, `bitvec!` and `expr!` macros and their constrains.
+///
+/// Tid IDs are optionally prefixed by `tid_name: `. If not, `tid_x` is set as Tid ID with incrementing `x` starting by `0`.
+/// ## Syntax
+/// Load: `var := Load from expr`, with a Variable `var` according to `variable!` macro and and expression `expr` according to `expr!` macro.
+///
+/// Store: `Store at expr_a := expr_b` with Expressions `expr_a` and `expr_b` according to `expr!` macro.
+///
+/// Assign: `var = expr` with a Variable `var` according to `var!` macro and an Expression `expr` according to `expr!` macro.
+/// ## Example
+/// ```rust
+///     use cwe_checker_lib::intermediate_representation::*;
+///     use cwe_checker_lib::def;
+///
+///     defs!["tid_x: Store at RSP:8 + 0x8:8 := RAX:8", "RSP:8 = RSP:8 + 0x8:8", "tid_z: RDI:8 := Load from RSP:8"];
+/// ```
+#[macro_export]
+macro_rules! defs {
+    [$($x:expr),*] => {{
+        let mut vec = vec![];
+        let mut _tid_suffix: u8 = 0;
+        $(
+            vec.push(parsing::parse_def($x, _tid_suffix));
+            _tid_suffix += 1;
+        )*
+        vec}
+
+    };
+}
+
+/// Creates a `Term<Def>` specified by the string slices. Utilizes `variable!`, `bitvec!` and `expr!` macros and their constrains.
+///
+/// Tid ID is optionally prefixed by `tid_name: `. If not, Tid ID `tid_0` is set.
+/// ## Syntax
+/// Load: `var := Load from expr`, with a Variable `var` according to `variable!` macro and and expression `expr` according to `expr!` macro.
+///
+/// Store: `Store at expr_a := expr_b` with Expressions `expr_a` and `expr_b` according to `expr!` macro.
+///
+/// Assign: `var = expr` with a Variable `var` according to `var!` macro and an Expression `expr` according to `expr!` macro.
+/// ## Example
+/// ```rust
+///     use cwe_checker_lib::intermediate_representation::*;
+///     use cwe_checker_lib::def;
+///
+///     def!["tid_x: Store at RSP:8 + 0x8:8 := RAX:8"];
+///     def!["RSP:8 = RSP:8 + 0x8:8"];
+/// ```
+#[macro_export]
+macro_rules! def {
+    ($x:expr) => {
+        parsing::parse_def($x, 0)
+    };
+}
+
+pub mod parsing {
+    //! Provides parsing functions for the macros defined in `macros.rs`.
+    //! This module hides the parsing functions and allows exposure of the macros only.
```

Comment:
To completely hide the parsing functions you should make the module private and add a `use parsing::*;` afterwards. This way the module would be completely inaccessible outside of the `macros` module.

Reply:
I am getting the `error failed to resolve: use of undeclared crate or module parsing
use of undeclared crate or module parsing` originating from the analysis' tests. This might be due to the fact, that the module `parsing` and the contained function is not in scope there but its call is "injected" by the macro.
Maybe leaving the module as `pub` is a more or less clean way to make the function usable everywhere? It will not be hidden, but maybe we can not do this at all?